src/minio/core/minio_client.py

```diff
@@ -81,3 +86,213 @@ async def test_connection(self) -> bool:
+                        await client.delete_objects(
+                            Bucket=bucket_name, Delete={"Objects": objects}
+                        )
+
+                await client.delete_bucket(Bucket=bucket_name)
+                logger.info(f"Deleted bucket: {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Failed to delete bucket {bucket_name}: {e}")
+            raise BucketOperationError(f"Bucket deletion failed: {e}") from e
+
+    async def put_object(self, bucket_name: str, key: str, body: bytes) -> bool:
+        """
+        Uploads an object to a specified bucket.
+
+        Args:
+            bucket_name: The name of the target bucket.
+            key: The object key (i.e., its name/path within the bucket).
+            body: The content of the object as bytes.
+
+        Returns:
+            True if the object was uploaded successfully.
+
+        Raises:
+            MinIOManagerError: If the upload operation fails.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.put_object(Bucket=bucket_name, Key=key, Body=body)
+                logger.info(f"Put object {key} in bucket {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Failed to put object {key} in bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object put failed: {e}") from e
+
+    async def get_object(self, bucket_name: str, key: str) -> bytes:
+        """
+        Retrieves an object from a bucket.
+
+        Args:
+            bucket_name: The name of the bucket containing the object.
+            key: The key of the object to retrieve.
+
+        Returns:
+            The content of the object as bytes.
+
+        Raises:
+            MinIOManagerError: If the object retrieval fails.
+        """
+        try:
+            async with self._get_client() as client:
+                response = await client.get_object(Bucket=bucket_name, Key=key)
+                body = await response["Body"].read()
+                logger.info(f"Got object {key} from bucket {bucket_name}")
+                return body
+        except Exception as e:
+            logger.error(f"Failed to get object {key} from bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object get failed: {e}") from e
+
+    async def delete_object(self, bucket_name: str, key: str) -> bool:
+        """
+        Deletes a single object from a bucket.
+
+        Args:
+            bucket_name: The name of the bucket containing the object.
+            key: The key of the object to delete.
+
+        Returns:
+            True if the object was deleted successfully.
+
+        Raises:
+            MinIOManagerError: If the object deletion fails.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.delete_object(Bucket=bucket_name, Key=key)
```

Comment:
What happens if the object doesn't exist? If it causes an exception to be thrown a noop might make more sense

Reply:
Same for delete bucket I guess

Reply:
If obj doesn't exist - it would error - which is fine to me. Otherwise we would need something like `object_exists` method - I think it's a bit overkill.

Reply:
hmmm - if object doesn't exist it would return False. Where did you see it would raise error if obj doesn't exist?

Reply:
Ok, I would prefer delete is successful if the object doesn't exist, but if that's what you want fair enough